Multiple changes for animated images

quentin-bettoum opened the pull request:
- Add a new `interframe_maxerror` parameter for GIFs which allow to increase compression of the frames of the GIF
- Update the `pages` option to allow the use from `-1` which allow to select all the pages of the animated image
- Some minor changes

quentin-bettoum commented on lib/image/options/write.ex:
```diff
@@ -240,10 +256,10 @@ defmodule Image.Options.Write do
 
   defp image_type_from("", suffix) do
-    {:ok, suffix}
+    {:ok, String.trim_leading(suffix, ".")}
```
This allows suffixes with or without a leading dot.
So both `suffix: ".jpg"` and `suffix: "jpg"` will work.

quentin-bettoum commented on lib/image/options/write.ex:
```diff
@@ -240,10 +256,10 @@ defmodule Image.Options.Write do
-  defp image_type_from(extname, _suffix) do
-    {:ok, extname}
+  defp image_type_from(path, _suffix) do
+    {:ok, path |> String.split(".") |> List.last() |> String.downcase()}
```
By downcasing the extension, there is no more error in case the output name is set to `IMG.JPG` for example.

quentin-bettoum commented:
I was wondering if it would make sense to change the default `:pages` parameter from `1` to `-1` meaning it would select all the pages of an animated image by default.

It would make sense since keeping animation from input to the output is probably the most common case, unlike picking only the first frame by default.

quentin-bettoum commented:
I reverted the changes made on the leading dot of the `:suffix` parameter.

kipcole9 commented on lib/image.ex:
```diff
@@ -925,6 +946,20 @@ defmodule Image do
+
+  * `:effort` is an integer to adjust the level of CPU
+    effort to reduce the file size.
+    The value must be in the range `1..10``, the default
```
`1..10`` -> `1..10`

kipcole9 commented:
Overall looks really great @quentin-bettoum - truly appreciated.  I think keeping `:suffix` with the included leading `.` is the most illustrative for me - it conveys the understanding its a file suffix more clearly that if it does not have the leading `.` in my opinion.  But I don't want to be dogmatic about it so additional conversation welcome - perhaps outside this round of PRs?

> I was wondering if it would make sense to change the default :pages parameter from 1 to -1 meaning it would select all the pages of an animated image by default.

That makes good sense to me!

quentin-bettoum commented:
Ok I fixed the typo and added the default `-1` for the `pages | n` parameter.

> Overall looks really great @quentin-bettoum - truly appreciated. I think keeping :suffix with the included leading . is the most illustrative for me - it conveys the understanding its a file suffix more clearly that if it does not have the leading . in my opinion. But I don't want to be dogmatic about it so additional conversation welcome - perhaps outside this round of PRs?

Yes it's not a big deal, I reverted it, the most important change about the extension was to allow recognition for uppercase extensions such as `image.JPG`, because in such case, the `is_jpg` would fail. And since in some case I would just reuse the input extension for the output, this would cause some errors for non-recognized options just because the extension was uppercase.

kipcole9 commented:
🥳🥳🥳🥳🥳

Thank you!
